staff/lab/mod-printer

```diff
@@ -0,0 +1,131 @@
+
+    if args.verbose:
+        print("Found CUPS classes: %s" % cups_classes)
+    
+    # Override with specified classname if provided and valid
+    if args.classname:
+        if args.classname in cups_classes:
+            cups_classes = [args.classname]
+            if args.verbose:
+                print("Overriding CUPS classes with parameter: %s" % args.classname)
+        else:
+            print("ERROR: Specified class %s could not be found in available classes %s" % (args.classname, cups_classes))
+            return 1
+    
+    # Perform pre-call logging and input processing
+    printer = args.printer.lower()
+    if args.verbose:
+        print("Printer to perform action on: %s" % printer)
+    action_flag = "-r" if args.action == "remove" else "-c"
+    if args.verbose:
+        print("Action to be performed (and flag): %s (%s)" % (args.action, action_flag))
+        
+    # Call lpadmin command to perform addition/removal
+    for classname in cups_classes:
+        action_cmd = ["lpadmin", "-p", printer + "-" + classname, action_flag, classname]
+        try:
+            subprocess.run(action_cmd, check=True, stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            print("ERROR: lpstat process exited with exit code %i" % e.returncode)
+            return e.returncode
+        else:
+            action_out = "added to" if args.action == "add" else "removed from"
+            print("Printer %s was successfully %s %s" % (printer, action_out, classname))
+            
+
+def list_printers(args):
+    """Perform internal call to CUPS CLI to list printer status(es) (and potentially jobs)."""
+    # Initialize base lpstat command
+    lpstat_cmd = ["lpstat", "-c"]
+
+    # Add on to lpstat command with arguments
+    if args.classname:
+        if args.verbose:
+            print("Limiting results to only class %s" % args.classname)
+        lpstat_cmd.append(args.classname)
+    if args.jobs:
+        if args.verbose:
+            print("Including jobs in stdout")
+        lpstat_cmd.append("-o")
+
+    if args.verbose:
+        print("Calling %s to list printer status" % lpstat_cmd)
+
+    try:
+        # Call lpstat - prints members of the class and can 
+        # be configured to include jobs as well. No way to 
+        # check for printers not in classes.
+        lpstat_out = subprocess.run(lpstat_cmd, check=True, stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+    except subprocess.CalledProcessError as e:
+        print("ERROR: lpstat process exited with exit code %i" % e.returncode)
+        return e.returncode
+
+    # Parse output to fit printer arg, or send directly to stdout
+    status_msgs = lpstat_out.stdout.decode("utf-8")
+    if args.printer:
+        status_lines = status_msgs.split("\n")
+        for line in status_lines[:]:
+            if args.printer not in line and "members of class" not in line:
+                status_lines.remove(line)
```

Comment:
No need to mutate the list, since you aren't using it anywhere else. You can print lines while looping, and skip printing the ones that don't fit the criteria.